### ISSUE TITLE
Cancel the current request

### ISSUE THREAD

**Issue by d3w3r** [danirod/cartero#82]
Actually if a make a request that takes so much time the current tab of the request is blocked and there is no way to abort the request, i found this problem calling one API that has a bad implementation so i put some images:

![image](https://github.com/user-attachments/assets/7608f76c-0017-49d5-a9e4-ee0de6a9547b)

> The solution could be a cancel button for the request.

**Comment by d3w3r** [comment]
Trying this argument in the url was a bad idea 😆 

![image](https://github.com/user-attachments/assets/c813422a-f7e7-4f49-818f-c49bbd4baf68)

**Comment by danirod** [comment]
Yeah, I am aware that the application currently lacks a way to cancel requests. I've suffered this and the only way I got to regain control of the program was to open a new tab and continue working there.

The fact that it's still not possible to set a predictable timeout in Cartero also makes the thing more slow.

I think that the least Cartero could do right now is to offer a button to stop the current request somewhere, either replacing the 'Send' button while a request is in progress, or covering the entire response pane. The first approach is faster to implement, the other approach could open more possibilities (displaying error messages in a pane, without using toasts, which is clunky)

**Comment by d3w3r** [comment]
Maybe for a timeout what about a preset value for all request and at the same time that inside each request the user can set manually a different timeout.

**Comment by danirod** [comment]
That's a really good idea, in fact. Maybe each request (and folder and collection) could have its own variable value to override the global settings or the parent settings.

**Comment by danirod** [comment]
Self-assigning this because now that #155 has been merged, it seems like a good moment to implement this. The code for the HTTP client and the UI is clean and clear, and it is possible to add new buttons with new behaviors.

Since this issue was first created, Cartero has gained new features (such as real timeout settings or better error reporting). However, still it would be nice to be able to stop a request that is taking too much.

It is important to keep in mind that cancelling a request is not safe. There is no guarantee that cancelling a request will stop any server-side effects. The intention of the Cancel button is to drop the socket connection at the TCP level, but there is no guarantee that the server will honor this or even will even know that it's been cancelled. Therefore, if the requested endpoint has to do things like update a database, send e-mails or stuff like that, it should be assumed that the side effects will be executed anyway.

Therefore, cancelling the current request will only have three effects:

* Resetting the UI to the previous state, allowing the user to edit data again.
* Showing a "cancelled" error in the response panel, similar to the ones added recently in #129.
* Closing the underlying socket connection to drop whatever the server responds.

**Comment by danirod** [comment]
Implemented in #160.
